pkg/core/native/native_neo.go

```diff
@@ -286,13 +290,66 @@ func (n *NEO) OnPersist(ic *interop.Context) error {
 func (n *NEO) PostPersist(ic *interop.Context) error {
 	gas := n.GetGASPerBlock(ic.DAO, ic.Block.Index)
 	pubs := n.GetCommitteeMembers()
-	index := int(ic.Block.Index) % len(ic.Chain.GetConfig().StandbyCommittee)
+	committeeSize := len(ic.Chain.GetConfig().StandbyCommittee)
+	index := int(ic.Block.Index) % committeeSize
 	gas.Mul(gas, big.NewInt(committeeRewardRatio))
 	n.GAS.mint(ic, pubs[index].GetScriptHash(), gas.Div(gas, big.NewInt(100)))
+
+	if shouldUpdateCommittee(ic.Block.Index, ic.Chain) {
+		var voterReward = big.NewInt(voterRewardRatio)
+		voterReward.Mul(voterReward, gas)
+		voterReward.Mul(voterReward, big.NewInt(storeFactor*int64(committeeSize)))
+		var validatorsCount = ic.Chain.GetConfig().ValidatorsCount
+		voterReward.Div(voterReward, big.NewInt(int64(committeeSize+validatorsCount)))
+		voterReward.Div(voterReward, big.NewInt(100))
+
+		cs := n.committeeWithVotes.Load().(keysWithVotes)
+		for i := range cs {
+			if cs[i].Votes.Sign() > 0 {
+				tmp := big.NewInt(1)
+				if i < validatorsCount {
+					tmp = big.NewInt(2)
+				}
+				tmp.Mul(tmp, voterReward)
+				tmp.Div(tmp, cs[i].Votes)
+
+				var key = makeVoterKey([]byte(cs[i].Key))
+				var reward = n.getGASPerVote(ic.DAO, key, ic.Block.Index+1)
+				tmp.Add(tmp, &reward[0])
+
+				var buf = make([]byte, 4)
```

Comment:
Better preallocate it once before all iterations.

Reply:
And this is probably true for most of other local variables here.

Reply:
Fixed.